app/src/main/java/org/dhis2/Bindings/DateExtensions.kt

```diff
@@ -12,10 +12,10 @@ import org.joda.time.Interval
 import org.joda.time.LocalDate
 import org.joda.time.Minutes
 
-fun Date?.toDateSpan(context: Context): String {
-    return if (this == null) {
-        ""
-    } else {
+fun Date?.toDateSpan(context: Context): String = when {
+    this == null -> ""
+    this.after(Date()) -> SimpleDateFormat("d/M/yyyy", Locale.getDefault()).format(this)
+    else -> {
```

Comment:
Do you think we could have unit test for this? @mmmateos

Reply:
Sure